package/MDAnalysis/analysis/rdf.py

```diff
@@ -261,6 +265,13 @@ def _single_frame(self):
             mask = np.where(idxA != idxB)[0]
             dist = dist[mask]
 
+        if self.ignore_same_residue:
```

Comment:
The exclusion_block argument achieves something similar right? I like the idea of a way to easily exclude based on same residue but is there a way to make it more general? eg

```
exclude_same_attr="resindices"
```

or failing that perhaps use the same code path as `_exclusion_block`? You could figure out the exclusion block from the indices possibly. 

What does @orbeckst think?

Reply:
tbh I am not sure exactly what _exclusion_block is doing. I'm pretty confused by the code. 

Happy to generalize to other properties, my only fear is that it makes it easier to mis-use. We might want to have a short list of allowed options. What else do you think would be worth including?

Reply:
It's been a while, but I think `exclusion_block` is used to tile out self contributions from molecules/residues of the same size.  

This then won't work if your residues are differently sized, so this approach here is preferable.

Reply:
Got it. In that case I can push @hmacdope's general purpose exclusion idea.